perception/map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -585,6 +585,54 @@ bool hasPotentialToReach(
   return false;
 }
 
+template <typename T>
+std::unordered_set<std::string> removeOldObjectsHistory(
+  const double current_time, const double buffer_time,
+  std::unordered_map<std::string, std::deque<T>> & target_objects)
+{
+  std::unordered_set<std::string> invalid_object_id;
+  for (auto iter = target_objects.begin(); iter != target_objects.end(); ++iter) {
+    const std::string object_id = iter->first;
+    std::deque<T> & object_data = iter->second;
+
+    // If object data is empty, we are going to delete the buffer for the obstacle
+    if (object_data.empty()) {
+      invalid_object_id.insert(object_id);
+      continue;
+    }
+
+    const double latest_object_time = rclcpp::Time(object_data.back().header.stamp).seconds();
+
+    // Delete Old Objects
+    if (current_time - latest_object_time > buffer_time) {
+      invalid_object_id.insert(object_id);
+      continue;
+    }
+
+    // Delete old information
+    while (!object_data.empty()) {
```

Comment:
this is not a new function, but a slightly modified version of `removeOldObjectsHistory` to accommodate crosswalk user history. I agree that it will look better with remove_if, but I think it is not a good idea to make refactoring while adding features.

Reply:
@dkoldaev but these are new added lines, so it is not a refactoring of old code? Even so, function logic is not altered so I really suggest you do it on this PR

Reply:
@danielsanchezaran I also am not sure how to implement an early return in case of remove if. The predicate is bool, so it will go through all of the iterator list as I understand. The early return is possible, because of while loop and sorted structure of the object data... I would keep it as is for now and address it in a separate PR if you want to improve here

Reply:
Please see my latest comment on the issue